packages/hg/src/browser/hg-contribution.ts

```diff
@@ -187,14 +187,19 @@ export class HgContribution implements CommandContribution, MenuContribution, Ta
             commandId: HG_COMMANDS.TRACK.id,
             when: 'scmProvider == hg && scmResourceGroup == untracked'
         });
+        registerResourceAction('1_modification', {
+            commandId: HG_COMMANDS.DISCARD.id,
+            when: 'scmProvider == hg && scmResourceGroup == changed'
+        });
```

Comment:
Surely if we support 'discard' but do not support 'forget' then we should be doing it the same way as Git.  I.e. the user should delete the file first (typically in the Navigator).  There is no equivalent in Git to 'forget' (in the case where it applies to a committed file), and as files without changes do not show up in the Source Control view, such actions would have to be available in the navigator.  I think this change is ok, but I just want to be sure because I know Joe wants things to work the same in Git and Hg where it makes sense to treat them the same.

Reply:
With this change the `forget` action is completely removed, so, if I understand correctly what you mean, the interface should be the same you get on Git.

Reply:
Yes, I see now the Git implementation deletes the file if it's not in the index, so we are fine.